client/go/outline/parse_test.go

```diff
@@ -36,50 +38,92 @@ func parseFirstHopAndTunnelConfigJSON(t *testing.T, jsonStr string) parsedTunnel
 	return parsed
 }
 
+// matchClientConfig is a helper to match the clientConfigString against the `client` field in the firstHopAndTunnelConfigString JSON.
+func matchClientConfig(t *testing.T, clientConfigString string, firstHopAndTunnelConfigString string) {
+	t.Helper()
+
+	var firstHopAndTunnelConfigValue map[string]any
+	require.NoError(t, json.Unmarshal([]byte(firstHopAndTunnelConfigString), &firstHopAndTunnelConfigValue))
+
+	var expected, actual map[string]any
+	require.NoError(t, yaml.Unmarshal([]byte(clientConfigString), &expected))
+	require.NoError(t, yaml.Unmarshal([]byte(firstHopAndTunnelConfigValue["client"].(string)), &actual))
+	require.Equal(t, expected, actual)
+}
+
+// matchTransportConfig is a helper to match the transportConfigString against the `client.transport` field in the firstHopAndTunnelConfigString JSON.
+func matchTransportConfig(t *testing.T, transportConfigString string, firstHopAndTunnelConfigString string) {
+	t.Helper()
+
+	var expected any
+	require.NoError(t, yaml.Unmarshal([]byte(transportConfigString), &expected))
```

Comment:
Just wanna double confirm, does `yaml.Unmarshal` correctly handle special characters?

Reply:
Yes, we have been using that for a while, and the tests exercise that.